### PR TITLE
Switch component

### PR DESCRIPTION
**Pull request by RustemYuzlibaev**

### CODE REVIEW

**Review by 0x009922 — CHANGES_REQUESTED** [review]
Everything is great. But I miss the main feature of the component - control it via `v-model` )

**Comment by RustemYuzlibaev** [comment]
> Everything is great. But I miss the main feature of the component - control it via `v-model` )

Decided to implement with Vue3 approach using **defineEmits** as in Alert component
If I get correctly we can control this way as well as via v-model?

**Comment by 0x009922** [comment]
@RustemYuzlibaev, I think the best way to implement it is to use `modelValue` as prop and `update:modelValue` as event, so it will be possible to use switch component like this:

```html
<SSwitch v-model="checked" />
```

**Review by 0x009922 — CHANGES_REQUESTED** [review]
- Use `modelValue` & `update:modelValue` (also look at `useVModel()` from [VueUse](https://vueuse.org/core/usevmodel/))
- Add Cypress test case with modeling check

File: packages/ui/src/components/Switch/SSwitch.vue [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+const emit = defineEmits<(event: 'update:modelValue', value: boolean) => void>()
+const model = useVModel(props, 'modelValue', emit)
+</script>
+
+<template>
+  <div class="s-switch">
+    <input :id="id" v-model="model" type="checkbox" :disabled="disabled" class="s-switch-button" />
+    <label :for="id" class="s-switch-label">{{ label }}</label>
+  </div>
```
Comment (0x009922) [inline packages/ui/src/components/Switch/SSwitch.vue:39]: According to BEM, isn't it correct to name sub-elements like `s-switch__button` & `s-switch__label`?

File: packages/ui/cypress/component/SSwitch.spec.cy.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+})
+
+it('SSwitch - renders with specified label', () => {
+  mount(SSwitch, { props: { id: 'id', label: 'Label' } })
+
+  cy.contains('Label')
+})
+
+it('SSwitch - renders unchecked switch when false prop is passed', () => {
+  mount(SSwitch, { props: { id: 'id', checked: false } })
```
Comment (0x009922) [inline packages/ui/cypress/component/SSwitch.spec.cy.ts:20]: Component doesn't have a prop with name `checked`.

**Review by 0x009922 — CHANGES_REQUESTED** [review]
Can you please resolve comments and add Cypress test for ID binding (ensure that id for label & input are the same)?

File: packages/ui/src/components/Alert/SAlert.vue [hunk 3]
```diff
@@ -49,14 +45,8 @@ function onClickClose() {
...
-    <div
-      v-if="showCloseBtn"
-      class="s-alert__close-wrapper"
-    >
-      <button
-        data-testid="close-btn"
-        @click="onClickClose"
-      >
+    <div v-if="showCloseBtn" class="s-alert__close-wrapper">
+      <button data-testid="close-btn" @click="onClickClose">
```
Comment (0x009922) [inline packages/ui/src/components/Alert/SAlert.vue:49]: Looks like formatting is broken :<

File: packages/ui/src/components/Alert/SAlert.vue [hunk 4]
```diff
@@ -26,11 +26,7 @@ function onClickClose() {
 </script>
 
 <template>
-  <div
-    class="s-alert"
-    :data-status="status"
-    :data-inline="inline"
-  >
+  <div class="s-alert" :data-status="status" :data-inline="inline">
```
Comment (0x009922) [inline packages/ui/src/components/Alert/SAlert.vue:29]: Here formatting is broken too

File: packages/ui/src/components/Switch/SSwitch.vue [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+const emit = defineEmits<(event: 'update:modelValue', value: boolean) => void>()
+const model = useVModel(props, 'modelValue', emit)
+</script>
+
+<template>
+  <div class="s-switch">
+    <input :id="id" v-model="model" type="checkbox" :disabled="disabled" class="s-switch__button" />
+    <label :for="id" class="s-switch__label">{{ label }}</label>
+  </div>
```
Comment (0x009922) [inline packages/ui/src/components/Switch/SSwitch.vue:39]: And here I see some formatting problems

**Review by 0x009922 — CHANGES_REQUESTED** [review]
Fix formatting please

**Review by 0x009922 — APPROVED** [review]

**Review by stefashkaa — APPROVED** [review]

**Pull request merged**